### PR TITLE
Cookie expiry fixes

### PR DESCRIPTION
**Pull request by ShivanKaul**
Resolves https://github.com/brave/brave-browser/issues/15048

1. Removes existing patches for max cookie expiry
2. Set overall lifetime max (for HTTP cookies) to be 6 months in CanonicalCookie::ValidateAndAdjustExpiryDate
3. Add browser tests for the 3 ways in which cookies can be set: HTTP, `document.cookie` and [Cookie Store API](https://developer.mozilla.org/en-US/docs/Web/API/Cookie_Store_API)
4. Set max expiry for JS cookies in RestrictedCookieManager

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Same test plan as https://github.com/brave/brave-core/pull/1905#issue-419029258

### CODE REVIEW

File: chromium_src/services/network/cookie_expiry_browsertest.cc
Comment (goodov) [inline chromium_src/services/network/cookie_expiry_browsertest.cc:26]: please don't put tests into `chromium_src` dir. `brave/browser/brave_shields` is fine for this browsertest.

File: chromium_src/services/network/cookie_expiry_browsertest.cc
Comment (goodov) [inline chromium_src/services/network/cookie_expiry_browsertest.cc:68]: `CHECK` -> `ASSERT_TRUE`

File: chromium_src/services/network/cookie_expiry_browsertest.cc
Comment (goodov) [inline chromium_src/services/network/cookie_expiry_browsertest.cc:82]: no need to do the manual `domAutomationController.send()` stuff. `EvalJs/ExecJs` supports promises and async calls.
https://github.com/chromium/chromium/blob/396799b65ad67a7ea10bbdeb1d75029d297bb520/content/public/test/browser_test_utils.h#L803-L809

File: chromium_src/services/network/cookie_expiry_browsertest.cc
Comment (goodov) [inline chromium_src/services/network/cookie_expiry_browsertest.cc:112]: please don't pollute the global namespace. this is usually "fun" to debug when a similar function appears in another test, but with a slightly different logic.

pls either move all these function into anonymous ns or move them into a test class.

File: chromium_src/services/network/cookie_expiry_browsertest.cc
Comment (goodov) [inline chromium_src/services/network/cookie_expiry_browsertest.cc:133]: what is this for?

File: chromium_src/services/network/cookie_expiry_browsertest.cc
Comment (goodov) [inline chromium_src/services/network/cookie_expiry_browsertest.cc:141]: is `ToDeltaSinceWindowsEpoch()` necessary? `Time - Time` should return `TimeDelta`.

File: chromium_src/services/network/restricted_cookie_manager.cc
Comment (goodov) [inline chromium_src/services/network/restricted_cookie_manager.cc:43]: static or namespace {}.

File: patches/net-cookies-canonical_cookie.cc.patch
Comment (goodov) [inline patches/net-cookies-canonical_cookie.cc.patch:9]: `BRAVE_CANONICAL_COOKIE_VALIDATE_AND_ADJUST_EXPIRY_DATE`

**Review by fmarier — APPROVED** [review]
Security review not needed since this is maintaining the same policy as before, just fixing a regression.

**Review by fmarier — CHANGES_REQUESTED** [review]
Actually, did you forget to `git add` the browsertest you added?

**Comment by ShivanKaul** [comment]
Moved the test and didn't add it again 😅

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (fmarier) [inline browser/brave_shields/cookie_expiry_browsertest.cc:77]: This is not quite a full end-to-end test of the `Cookie` header, right?

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (fmarier) [inline browser/brave_shields/cookie_expiry_browsertest.cc:138]: I'd suggest adding a test case for a cookie expiring in 2 days (to ensure the expiry is not extended to 7 days).

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (fmarier) [inline browser/brave_shields/cookie_expiry_browsertest.cc:168]: And here a test case for 30 days would be nice.

File: chromium_src/services/network/restricted_cookie_manager.h
Comment (bridiver) [inline chromium_src/services/network/restricted_cookie_manager.h:17]: there's no need to make this a method on the class

Comment (ShivanKaul) [reply chromium_src/services/network/restricted_cookie_manager.h:17]: No strong feelings on my end, just that it seemed cleanest to keep this method together with the existing ones?

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (ShivanKaul) [reply browser/brave_shields/cookie_expiry_browsertest.cc:138]: This would need a time window, because there might be a gap of a few milliseconds between executing the script to set the cookie and the cookie getting stored. I'll set the window to be a few seconds to prevent flapping tests.

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (goodov) [inline browser/brave_shields/cookie_expiry_browsertest.cc:69]: `ASSERT_TRUE(content::ExecJs`

File: chromium_src/services/network/DEPS
Comment (goodov) [inline chromium_src/services/network/DEPS:2]: just revert the changes.

**Review by fmarier — APPROVED** [review]

File: chromium_src/net/cookies/canonical_cookie.cc
Comment (goodov) [inline chromium_src/net/cookies/canonical_cookie.cc:25]: we're currently reenabling unreachable-code warning, so to not break `master` when both PRs will be merged simultaneously, pls make this change like so:
```
if ((true))
  return BraveCanonExpiration(expiry_date, fixed_creation_date);
```
some details here if curious: https://github.com/brave/brave-core/pull/14904

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (goodov) [inline browser/brave_shields/cookie_expiry_browsertest.cc:29]: anonymous namespace pls for these constants.

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (goodov) [inline browser/brave_shields/cookie_expiry_browsertest.cc:186]: embedded test server has always enabled `/set-cookie` endpoint to do exactly this.

impl:
https://github.com/chromium/chromium/blob/8def63071324792533c0db180cd35c309a9c623a/net/test/embedded_test_server/default_handlers.cc#L239-L257

example:
https://github.com/chromium/chromium/blob/76f28918e8af7b367f384433f9d55f4e9b76b1b0/chrome/browser/content_settings/content_settings_browsertest.cc#L534-L536

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (goodov) [inline browser/brave_shields/cookie_expiry_browsertest.cc:37]: make_unique

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (goodov) [inline browser/brave_shields/cookie_expiry_browsertest.cc:62]: `ASSERT_TRUE(content::ExecJs(`

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (goodov) [inline browser/brave_shields/cookie_expiry_browsertest.cc:200]: why `magnitude()` (an absolute value)? this might hide a bug.

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (goodov) [inline browser/brave_shields/cookie_expiry_browsertest.cc:228]: @ShivanKaul did you do a sanity check to make sure that the real time forwarding on a machine actually prevents the cookie to be returned to a website?
I might be a little paranoid, but if our changes are in the wrong place, then `ExpiryDate` can say right things, but the real scenario might fail.

Comment (ShivanKaul) [reply browser/brave_shields/cookie_expiry_browsertest.cc:228]: Yep!

**Review by goodov — APPROVED** [review]

**Comment by goodov** [comment]
do a rebase please before merging.

**Review by bridiver — APPROVED** [review]

File: browser/brave_shields/cookie_expiry_browsertest.cc
Comment (antonok-edm) [inline browser/brave_shields/cookie_expiry_browsertest.cc:3]: @ShivanKaul fyi this file has an incorrect license header

**Pull request merged**
